session.go

```diff
@@ -154,16 +157,31 @@ func SetSessionStateCallback(f SessionStateCallback) {
 	sessStateCallback = f
 }
 
-func SetKvReloadHandler(h KvReloadHandler) {
-	kvReloadHandler = h
+// SetKeyValuesReadyCallback designates a function to be called when the Device
+// Data Proxy is ready to be accessed.
+//
+// IMPORTANT: This callback function runs in a separate goroutine. Please take
+// care to handle any concurrency issues.
+func SetKeyValuesReadyCallback(f KeyValuesReadyCallback) {
+	keyValuesReadyCallback = f
 }
 
-func SetKvSetHandler(h KvSetHandler) {
-	kvSetHandler = h
+// SetKeyValueStoredCallback designates a function to be called whenever a
+// key-value pair has been added or updated by someone else.
+//
+// IMPORTANT: This callback function runs in a separate goroutine. Please take
+// care to handle any concurrency issues.
+func SetKeyValueStoredCallback(f KeyValueStoredCallback) {
+	keyValueStoredCallback = f
 }
 
-func SetKvDeleteHandler(h KvDeleteHandler) {
-	kvDeleteHandler = h
+// SetKeyValueDeletedCallback designates a function to be called whenever a
+// key-value pair has been deleted by someone else.
+//
+// IMPORTANT: This callback function runs in a separate goroutine. Please take
+// care to handle any concurrency issues.
+func SetKeyValueDeletedCallback(f KeyValueDeletedCallback) {
```

Comment:
what if delete and set key value happens almost at the same time?

The callback may be called out of order?

For example, if DELETE -> SET "key" happens, then the order will be guaranteed?